core/src/main/java/org/jdbi/v3/core/mapper/MapEntryMapper.java

```diff
@@ -0,0 +1,164 @@
+    }
+
+    private final RowMapper<K> keyMapper;
+    private final RowMapper<V> valueMapper;
+
+    @SuppressWarnings("unchecked")
+    static RowMapperFactory factory() {
+        return (type, config) -> {
+            if (type instanceof ParameterizedType && getErasedType(type).equals(Map.Entry.class)) {
+                Type keyType = resolveType(KEY_PARAM, type);
+                Type valueType = resolveType(VALUE_PARAM, type);
+
+                RowMapper<?> keyMapper = getKeyMapper(keyType, config);
+                RowMapper<?> valueMapper = getValueMapper(valueType, config);
+
+                return Optional.of(new MapEntryMapper(keyMapper, valueMapper));
+            }
+            return Optional.empty();
+        };
+    }
+
+    private static RowMapper<?> getKeyMapper(Type keyType, ConfigRegistry config) {
+        String column = config.get(Config.class).getKeyColumn();
+        if (column == null) {
+            return config.get(RowMappers.class)
+                    .findFor(keyType)
+                    .orElseThrow(() -> new NoSuchMapperException("No row mapper registered for map key " + keyType));
+        } else {
+            return config.get(ColumnMappers.class)
+                    .findFor(keyType)
+                    .map(mapper -> new SingleColumnMapper<>(mapper, column))
+                    .orElseThrow(() -> new NoSuchMapperException("No column mapper registered for map key " + keyType + " in column " + column));
+        }
+    }
+
+    private static RowMapper<?> getValueMapper(Type keyType, ConfigRegistry config) {
+        String column = config.get(Config.class).getValueColumn();
+        if (column == null) {
+            return config.get(RowMappers.class)
+                    .findFor(keyType)
+                    .orElseThrow(() -> new NoSuchMapperException("No row mapper registered for map value " + keyType));
+        } else {
+            return config.get(ColumnMappers.class)
+                    .findFor(keyType)
+                    .map(mapper -> new SingleColumnMapper<>(mapper, column))
+                    .orElseThrow(() -> new NoSuchMapperException("No column mapper registered for map value " + keyType + " in column " + column));
+        }
+    }
+
+    private MapEntryMapper(RowMapper<K> keyMapper, RowMapper<V> valueMapper) {
+        this.keyMapper = keyMapper;
+        this.valueMapper = valueMapper;
+    }
+
+    @Override
+    public Map.Entry<K, V> map(ResultSet rs, StatementContext ctx) throws SQLException {
+        return immutableEntry(keyMapper.map(rs, ctx), valueMapper.map(rs, ctx));
+    }
+
+    @Override
+    public RowMapper<Map.Entry<K, V>> specialize(ResultSet rs, StatementContext ctx) throws SQLException {
+        return new MapEntryMapper<>(keyMapper.specialize(rs, ctx), valueMapper.specialize(rs, ctx));
+    }
+
+    /**
+     * Configuration class for MapEntryMapper.
+     */
+    public static class Config implements JdbiConfig<Config> {
+        public Config() {
```

Comment:
This is required by ConfigRegistry--all JdbiConfig instances must have a public no-arg constructor, so the registry can create a root config object on demand.